dataclients/kubernetes/kube.go

```diff
@@ -901,28 +914,22 @@ func (c *Client) ingressToRoutes(items []*ingressItem) ([]*eskip.Route, error) {
 			// currently handled as mandatory
 			host := []string{"^" + strings.Replace(rule.Host, ".", "[.]", -1) + "$"}
 
-			// add extra routes from optional annotation
-			for idx, r := range extraRoutes {
-				route := *r
-				route.HostRegexps = host
-				route.Id = routeIDForCustom(i.Metadata.Namespace, i.Metadata.Name, route.Id, rule.Host, idx)
-				hostRoutes[rule.Host] = append(hostRoutes[rule.Host], &route)
-			}
-
 			// update Traffic field for each backend
 			computeBackendWeights(backendWeights, rule)
 
 			for _, prule := range rule.Http.Paths {
-				if prule.Backend.Traffic > 0 {
-					pathMode := c.pathMode
-					pathModeString := i.Metadata.Annotations[pathModeAnnotationKey]
-					if pathModeString != "" {
-						var err error
-						if pathMode, err = ParsePathMode(pathModeString); err != nil {
-							return nil, err
-						}
-					}
+				pathMode := c.getPathMode(i)
+
+				// add extra routes from optional annotation
+				for idx, r := range extraRoutes {
+					route := *r
+					route.HostRegexps = host
+					route.Id = routeIDForCustom(i.Metadata.Namespace, i.Metadata.Name, route.Id, rule.Host, idx)
+					setPath(pathMode, &route, prule.Path)
```

Comment:
What do you think should be done?
In my opinion it should only add it as predicate and not overwrite the path rule.

Reply:
@aryszka we should document the behaviour and if it is a broken route, because of multiple not matching path related predicates, then it's a broken route and we should somehow report this. I think an error log should be enough.

A user can also specify a broken combination of predicates, which will not match anything and we do not stop them doing it.

Reply:
true that it's possible to create invalid set of predicates, valid argument.